exercises/twelve-days/example.js

```diff
@@ -0,0 +1,55 @@
+
+  (function() {
+    instance.verseList = [
+      'On the first day of Christmas my true love gave to me, a Partridge in a Pear Tree.',
+      'On the second day of Christmas my true love gave to me, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the third day of Christmas my true love gave to me, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the fourth day of Christmas my true love gave to me, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the fifth day of Christmas my true love gave to me, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the sixth day of Christmas my true love gave to me, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the seventh day of Christmas my true love gave to me, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the eighth day of Christmas my true love gave to me, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the ninth day of Christmas my true love gave to me, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the tenth day of Christmas my true love gave to me, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the eleventh day of Christmas my true love gave to me, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+      'On the twelfth day of Christmas my true love gave to me, twelve Drummers Drumming, eleven Pipers Piping, ten Lords-a-Leaping, nine Ladies Dancing, eight Maids-a-Milking, seven Swans-a-Swimming, six Geese-a-Laying, five Gold Rings, four Calling Birds, three French Hens, two Turtle Doves, and a Partridge in a Pear Tree.',
+    ];
+  })();
+
+  instance.startFromZero = function(oneIndexArray) {
+    var newArray = oneIndexArray.map(function(item) { return item - 1; });
+    return newArray;
+  };
+
+  instance.singleVerse = function(verseIndex) {
+    var verse = this.verseList[verseIndex].concat('\n');
+    return verse;
+  };
+
+  instance.multiVerse = function(startIndex, endIndex) {
+    var filteredList = this.verseList.filter(function (verse, index) {
```

Comment:
This is not a change request, just to know your opinion. What about chaining a couple of `filter` methods?

```javascript
return this.verseList
  .filter(function(verse, index) { return index >= startIndex })
  .filter(function(verse, index) { return index <= endIndex })
  .join('\n\n')
  .concat('\n')
```

Reply:
Your suggestion reads much better. I think the filter method just needs the callback to return a "truthy" value or not to decide whether to keep the element. Returning the boolean result instead of  `return verse;` makes it easier to understand.